### ISSUE TITLE
Can't run one Answer in multiple threads

### ISSUE THREAD

**Issue by GoogleCodeExporter** [dlsgusrn7577/mockito#265]
```
I need to test a program that kicks off multiple threads up to a certain limit. 
 In particular, I want to test that this limit is enforced.  The code in the 
thread itself is not relevant to the test, so long as it's something that runs 
for long enough to ensure that multiple threads can get going at once.

I have mocked a suitable Callable object, and used doAnswer to give it 
something to run when called.  My issue is that Mockito won't kick off the 
Answer in multiple threads at once.  I believe the problem is in the answer 
method of StubbedInvocationMatcher - I think 

synchronized(answers) {
    return answers.size() == 1 ? 
        answers.peek().answer(invocation) : 
        answers.poll().answer(invocation);
        }

should read something like

Answer answer;
synchronized(answers) {
    answer = answers.size() == 1 ? 
        answers.peek() : answers.poll();
    }
answer.answer( invocation );

but I'm wondering whether there's a particular reason why it was written in the 
way that it was.  For example, does using Answer objects do something else 
that's not thread safe?

I am running Mockito 1.8.5, Java 1.6.0 build 24 on Windows XP.

Cheers,
David.



```

Original issue reported on code.google.com by `topsoftw...@gmail.com` on 31 May 2011 at 9:05
- Merged into: #190

**Comment by GoogleCodeExporter** [comment]
```
Makes sense. Isnt it fixed in trunk? If not how about a fork and a pull request 
;)
```

Original comment by `szcze...@gmail.com` on 1 Jun 2011 at 12:00

**Comment by GoogleCodeExporter** [comment]
```
Yep, actually it's a duplicate of issue 190, which you fixed roughly one year 
ago.
Note that it's fixed in 1.9.0-rc1
```

Original comment by `brice.du...@gmail.com` on 3 Oct 2011 at 9:52
- Changed state: **Duplicate**
